docs/documentation/release_notes/topics/24_0_0.adoc

```diff
@@ -1,3 +1,27 @@
+= Deprecation of 'legacy' cookie behavior
+
+As part of the https://privacysandbox.com/[Privacy Sandbox] effort, Keycloak is making changes to the way cookies are handled. Specifically we are preparing for the https://developers.google.com/privacy-sandbox/blog/cookie-countdown-2023oct[end of third-party cookies], and are looking to improve the consistency, privacy and security of how cookies are handled by default within Keycloak. This has several implications for how Keycloak is configured and used:
+
+== Removal of cookies suffixed in `_LEGACY`
+
+Since the introduction of the https://web.dev/articles/samesite-cookies-explained[`SameSite`] attribute on cookies provided by Keycloak, another set of 'legacy' cookies (name suffixed in `_LEGACY`) were also introduced in order to support older browsers that were https://www.chromium.org/updates/same-site/incompatible-clients/[not compatible] with this attribute. Since the `SameSite` attribute is now https://caniuse.com/same-site-cookie-attribute[widely supported], this behavior is now deprecated, so that only the cookies using the `SameSite` attribute remain.
+
+== `SameSite` attribute set to `Strict` by default
+
+All cookies provided by Keycloak will now have the `SameSite` attribute set to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#strict[`Strict`] by default. This means that cookies will only be sent in a first-party context, that is, requests originating from the same site that set the cookie. Some specific cookies that are required in a third party contexts, such as the session cookie are exempt from this behavior for now, but this will likely change in the future as using cookies in a third-party context will no longer supported by browsers.
+
+== Support for insecure contexts dropped
+
+Due to the aforementioned changes, we are officially dropping support for running Keycloak outside of a https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts[secure context]. You should therefore ensure that your Keycloak instance if always running on what is https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#when_is_a_context_considered_secure[considered a secure context]. As part of this, all cookies set by Keycloak will have the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#secure[`Secure`] attribute by default.
```

Comment:
Does this mean that some more update might need to be done in the documentation in relation to this? I can think of some update for the option `http-enabled` documented here https://www.keycloak.org/server/enabletls (as per my understanding `http-enabled` won't work correctly unless people are running on `localhost` or legacy cookies feature is enabled).

Reply:
> Does this mean that some more update might need to be done in the documentation in relation to this?

Yes, the documentation will have to be reviewed and modified in this regard. Once the implementation settles down that will be the next step.